app/src/renderer/connectors/lcdClientMock.js

```diff
@@ -891,6 +916,142 @@ module.exports = {
       deposit => deposit.depositer === address
     )
   },
+  async submitProposalDeposit({
+    proposal_id,
+    base_req: { name, sequence },
+    depositer,
+    amount
+  }) {
+    let results = []
+    let fromKey = state.keys.find(a => a.name === name)
+    let fromAccount = state.accounts[fromKey.address]
+    if (fromAccount == null) {
+      results.push(txResult(1, `Nonexistent account`))
+      return results
+    }
+    // check nonce
+    if (parseInt(fromAccount.sequence) !== parseInt(sequence)) {
+      results.push(
+        txResult(
+          2,
+          `Expected sequence "${fromAccount.sequence}", got "${sequence}"`
+        )
+      )
+      return results
+    }
+
+    let proposal = state.proposals.find(
+      proposal => proposal.proposal_id === proposal_id
+    )
+    if (!proposal) {
+      results.push(txResult(3, `Nonexistent proposal`))
+      return results
+    } else if (
+      proposal.proposal_status != `Pending` &&
+      proposal.proposal_status != `Active`
+    ) {
+      results.push(txResult(3, `Proposal #${proposal_id} already finished`))
+      return results
+    }
+
+    // update depositer's balance
+    let coin
+    let submittedDeposit = {
+      proposal_id,
+      depositer,
+      amount
+    }
+
+    // javascript's forEach doesn't support break, so using classic for loop...
+    for (let i = 0; i < amount.length; i++) {
+      coin = amount[i]
+      let depositCoinAmt = parseInt(coin.amount)
+      let coinBalance = fromAccount.coins.find(c => c.denom === coin.denom)
+
+      if (depositCoinAmt < 0) {
+        results.push(txResult(1, `Amount of ${coin.denom}s cannot be negative`))
+        return results
+      } else if (!coinBalance || coinBalance.amount < depositCoinAmt) {
+        results.push(txResult(1, `Not enough ${coin.denom}s in your account`))
+        return results
+      }
+
+      coinBalance.amount -= depositCoinAmt
+
+      // ============= TOTAL PROPOSAL's DEPOSIT =============
+      // Increment total deposit of the proposal
+      let depositIndex = proposal.total_deposit.findIndex(
```

Comment:
I think you would reduce the complexity to read the code a little by using `.find` instead of `.findIndex` as you can edit the found deposit and do not need to query it again via the index.
What was your thought process to use `.findIndex`?

Reply:
yeah I started using .find() but then I changed Idk why though... updated